### PR TITLE
feat: add ability to specify application root

### PR DESCRIPTION
**Pull request by wcatron**
Add ability to specify a working directory for running `electron-builder`. Helps when developing an electron application inside a monorepo. Should be none breaking. First action contribution, open to suggestions on everything. Not sure how to test the script, would be willing to try and document process.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -114,10 +120,10 @@ const runAction = () => {
 	}
 
 	log(`${release ? "Releasing" : "Building"} the Electron app…`);
-	run(
+	runIn(
 		`${useNpm ? "npx --no-install" : "yarn run"} electron-builder --${platform} ${
 			release ? "--publish always" : ""
-		}`,
+		}`, root
```
Comment (samuelmeuli) [inline index.js]: Shouldn't the rest of the commands (e.g. `npm install`) also be run in that directory? If so, we could simply extend the `run` function with a new parameter instead of creating a new `runIn` helper.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "action-electron-builder",
-	"version": "1.0.0",
+	"version": "1.1.0",
```
Comment (samuelmeuli) [inline package.json]: I'd like to do this change in a separate commit.

File: index.js [hunk 3]
```diff
@@ -77,6 +82,7 @@ const setEnvVariable = (name, value) => {
 const runAction = () => {
 	const platform = getPlatform();
 	const release = getEnvVariable("release") === "true";
+	const root = getEnvVariable("application_root") && null
```
Comment (samuelmeuli) [inline index.js]: What does `&& null` do here?

File: index.js (hunk 1 above)
Comment (wcatron) [reply index.js]: So it can depend on the kind of mono-repo setup you are running - I am not a monorepo expert or even an advocate. I believe I have what should be a common use case though. A mono-repo with separate packages for: the electron application, shared tooling, shared libraries, a web server (that uses the shared models), utilities, etc. The entire mono repo must be installed and built in order to build the main electron application.

Alternatively you could have a giant mono-repo with a lot of code that is not required for the electron application. That in my opinion would require a more custom build script than this generic use action.

File: index.js (hunk 3 above)
Comment (wcatron) [reply index.js]: It appears to set `root` to null if you _define_ the environment variable. I meant ` || null`... `getEnvVariable` appears to return `undefined` when the env variable isn't set. I wanted to explicitly set `root` to `null` since `undefined` is not technically an acceptable value for the `cwd` option.

File: index.js (hunk 1 above)
Comment (samuelmeuli) [reply index.js]: Ok, works for me. If somebody needs it, a second variable (e.g. `package_root`) could be introduced in the future, which would define where NPM/Yarn commands are run.

**Comment by samuelmeuli** [comment]
Looks good to me. Have you been able to test the action?
